### DISCUSSION TITLE
How to authenticate with certificate?

### DISCUSSION THREAD

**Discussion by Tim-2020** [microsoft/winget-cli#3428] [Configuration]
I have a goal to restrict access to the apps repository somehow. I have found that there are several parameters on server side that allows to set certificate and raise client authorization by certificate. On client side it is required to send certificate in header `X-ARR-ClientCert`. It works fine if I wrote my own client. But is there any way to set this certificate in winget client? Or is there any way to set special custom http header in the winget client?
Thanks in advance!

**Comment by Trenly** [comment]
Does this answer your question?
* https://github.com/microsoft/winget-cli/issues/1919

**Comment by denelon** [comment]
WinGet does support passing an http header to a REST source so that can be used any way you like in your implementation, but we are working on broad authentication and RBAC style controls at a per package level.

**Reply by Tim-2020** [reply]
And how to pass parameter into http header? I was able only to set "Windows-Package-Manager" header with options. How can I set custom header key?

**Reply by denelon** [reply]
That's the only supported header today. As a naive suggestion, I'd say to treat it like some kind of key/value pair so you can pass the string along to the server and a custom behavior on the server could interpret the value passed in.
